identity_credential/src/presentation/presentation.rs

```diff
@@ -143,3 +157,94 @@ where
     self.fmt_json(f)
   }
 }
+
+#[cfg(test)]
+mod tests {
+  use serde_json::json;
+  use std::error::Error;
+
+  use identity_core::common::Object;
+  use identity_core::convert::FromJson;
+
+  use crate::presentation::Presentation;
+
+  #[test]
+  fn test_presentation_deserialization() {
+    // Example verifiable presentation taken from:
+    // https://www.w3.org/TR/vc-data-model/#example-a-simple-example-of-a-verifiable-presentation
+    // with some minor adjustments (adding the `holder` property and shortening the 'jws' values).
+    assert!(Presentation::<Object>::from_json_value(json!({
+      "@context": [
+        "https://www.w3.org/2018/credentials/v1",
+        "https://www.w3.org/2018/credentials/examples/v1"
+      ],
+      "holder": "did:test:abc1",
+      "type": "VerifiablePresentation",
+      "verifiableCredential": [{
+        "@context": [
+          "https://www.w3.org/2018/credentials/v1",
+          "https://www.w3.org/2018/credentials/examples/v1"
+        ],
+        "id": "http://example.edu/credentials/1872",
+        "type": ["VerifiableCredential", "AlumniCredential"],
+        "issuer": "https://example.edu/issuers/565049",
+        "issuanceDate": "2010-01-01T19:23:24Z",
+        "credentialSubject": {
+          "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+          "alumniOf": {
+            "id": "did:example:c276e12ec21ebfeb1f712ebc6f1",
+            "name": [{
+              "value": "Example University",
+              "lang": "en"
+            }, {
+              "value": "Exemple d'Université",
+              "lang": "fr"
+            }]
+          }
+        },
+        "proof": {
+          "type": "RsaSignature2018",
+          "created": "2017-06-18T21:19:10Z",
+          "proofPurpose": "assertionMethod",
+          "verificationMethod": "https://example.edu/issuers/565049#key-1",
+          "jws": "eyJhb...dBBPM"
+        }
+      }],
+    }))
+    .is_ok());
+  }
+
+  #[test]
+  fn test_presentation_deserialization_without_credentials() {
+    // Deserializing a Presentation without `verifiableCredential' property is allowed.
+    assert!(Presentation::<()>::from_json_value(json!({
+      "@context": [
+        "https://www.w3.org/2018/credentials/v1",
+        "https://www.w3.org/2018/credentials/examples/v1"
+      ],
+      "holder": "did:test:abc1",
+      "type": "VerifiablePresentation"
+    }))
+    .is_ok());
+  }
+
+  #[test]
+  fn test_presentation_deserialization_with_empty_credential_array() {
+    assert_eq!(
+      Presentation::<()>::from_json_value(json!({
+        "@context": [
+          "https://www.w3.org/2018/credentials/v1",
+          "https://www.w3.org/2018/credentials/examples/v1"
+        ],
+        "holder": "did:test:abc1",
+        "type": "VerifiablePresentation",
+        "verifiableCredential": []
+      }))
+      .unwrap_err()
+      .source()
+      .unwrap()
+      .to_string(),
+      "empty verifiableCredential array in presentation"
+    );
```

Comment:
```suggestion
    assert_eq!(
      Presentation::<()>::from_json_value(json!({
        "@context": [
          "https://www.w3.org/2018/credentials/v1",
          "https://www.w3.org/2018/credentials/examples/v1"
        ],
        "holder": "did:test:abc1",
        "type": "VerifiablePresentation",
        "verifiableCredential": []
      }))
      .unwrap_err()
      .source()
      .unwrap()
      .to_string(),
      crate::Error::EmptyVerifiableCredentialArray.to_string()
    );
```

I think the serde error erases the type, right? So there's probably no better way to test.

Except you could compare against the string repr of the error, that would be more resilient.

Reply:
ahh yeah indeed, thanks for the suggestion :)